addons/project/models/project.py

```diff
@@ -568,6 +578,32 @@ def _read_group_stage_ids(self, stages, domain, order):
     # customer portal: include comment and incoming emails in communication history
     website_message_ids = fields.One2many(domain=lambda self: [('model', '=', self._name), ('message_type', 'in', ['email', 'comment'])])
 
+    # recurring task fields
+    allow_recurring_tasks = fields.Boolean(related="project_id.allow_recurring_tasks")
+    recurrency = fields.Boolean(string="Recurring Task", copy=False, help="Set up a repetition scheme for this task")  # , compute="_compute_recurrency_field", inverse="_inverse_recurrency_field", store=True
+    recurrency_interval = fields.Integer(string='Repeat Every', default=1, help="Repeat every X (weeks/months/years)", copy=False)
+    recurrency_type = fields.Selection([
+        ('weeks', 'Weeks'),
+        ('months', 'Months'),
+        ('years', 'Years')
+    ], string='Recurrence', default='months', help="Let the task automatically repeat at that interval", copy=False)
+    mo = fields.Boolean('Mon', copy=False)
+    tu = fields.Boolean('Tue', copy=False)
+    we = fields.Boolean('Wed', copy=False)
+    th = fields.Boolean('Thu', copy=False)
+    fr = fields.Boolean('Fri', copy=False)
+    sa = fields.Boolean('Sat', copy=False)
+    su = fields.Boolean('Sun', copy=False)
+    recurrency_stage_id = fields.Many2one('project.task.type', string="Recurrency Stage", domain ="[('project_ids', 'in', project_id)]", copy=False) # add domain to this to allow only stages from this project   
+    recurrency_end_type = fields.Selection([
+        ('forever', 'Forever'),
+        ('last_date', 'End date')
+    ], string='Recurrence Termination', default='forever', copy=False)
+    recurrency_until_date = fields.Date('Repeat Until', copy=False)
+    recurring_parent_id = fields.Many2one('project.task', string="Recurring Parent Task", copy=False, readonly=True)
+    recurring_child_ids = fields.One2many('project.task', 'recurring_parent_id', copy=False)
```

Comment:
One2Many fields are not copied by default ;)
https://github.com/odoo/odoo/blob/4ad7a99df0a25122cf6872f880e07b6148c82ea8/odoo/fields.py#L2781-L2787